### PR TITLE
Ignore 409 errors when initializing tile storage

### PR DESCRIPTION
**Pull request by paulius-valiunas**
A race condition can occur when multiple people try to open the same iModel at the same time. Since `@itwin/object-storage` doesn't have good error handling, this PR simply ignores 409 errors that come from Azure blob storage.

### CODE REVIEW

File: core/backend/src/TileStorage.ts [hunk 1]
```diff
@@ -44,7 +44,12 @@ export class TileStorage {
     if (this._initializedIModels.has(iModelId))
       return;
     if (!(await this.storage.baseDirectoryExists({ baseDirectory: iModelId }))) {
-      await this.storage.createBaseDirectory({ baseDirectory: iModelId });
+      try {
+        await this.storage.createBaseDirectory({ baseDirectory: iModelId });
+      } catch (e: any) {
+        if(e.statusCode !== 409)
```
Comment (nick4598) [inline core/backend/src/TileStorage.ts:50]: Is it worth documenting what a 409 is and why it might occur?

Comment (pmconne) [reply core/backend/src/TileStorage.ts:50]: Yes.

Comment (paulius-valiunas) [reply core/backend/src/TileStorage.ts:52]: It's a standard HTTP status code, but I added a comment explaining why this happens

**Review by pmconne — APPROVED** [review]

**Comment by paulius-valiunas** [comment]
@mergifyio backport release/4.4.x

**Pull request merged**